src/test/scala/query/MergeQuerySpec.scala

```diff
@@ -0,0 +1,159 @@
+package test.query
+
+import com.originate.scalypher.path.AnyNode
+import com.originate.scalypher.path.CypherNode
+import com.originate.scalypher.path.KindRelationship
+import com.originate.scalypher.MergeQuery
+
+import org.scalatest._
+
+class MergeQuerySpec extends WordSpec with Matchers {
+
+  val startNode = CypherNode("label1")
+  val createRelationship = KindRelationship("label")
+  val endNode = CypherNode("label2")
+  val matchPath = startNode -- endNode
+  val createPath = startNode -- createRelationship -- endNode
+  val where = startNode.property("id") <> "test"
+
+  "only a merge path is given" must {
+
+    "produce a query that merges a node" in {
+      val create = MergeQuery(startNode)
+      create.toQuery shouldBe "MERGE (:label1)"
+    }
+
+  }
+
+  "a match and merge path is given" must {
+
+    "use simple identifiers for references in the merge path" in {
+      val create = matchPath merge createPath
+      create.toQuery shouldBe "MATCH (a1:label1)--(a2:label2) MERGE (a1)-[:label]-(a2)"
+    }
+
+    "include a return statement if provided" in {
+      val create = matchPath merge createPath returnDistinct startNode
+      create.toQuery shouldBe """MATCH (a1:label1)--(a2:label2) MERGE (a1)-[:label]-(a2) RETURN DISTINCT a1"""
+    }
+
+    "add identifier to reference if it appears in the return statement" in {
+      val create = matchPath merge createPath returnDistinct createRelationship
+      val query = create.toQuery
+      query should include (")-[a1:label]-(")
```

Comment:
that would suffice but is less specific. including the `)-` and `-(` ensures that i'm matching a relationship that is placed in between two nodes that is undirected.